frontend/lib/src/components/transaction_component/transaction_component.dart

```diff
@@ -160,6 +162,26 @@ class TransactionComponent extends StatelessWidget {
                     height: 32,
                   ),
                 ],
+                if (type == TransactionTypes.CREDIT) ...[
+                  Dropdown(
+                    labelText: 'Category',
+                    items: creditCategory(),
+                    selectedValue:
+                        transactionController.selectedTransactionCategory,
+                    onChanged: (value) {
+                      transactionController.setSelectedTransactionCategory =
+                          value.toString();
```

Comment:
isnt the value already a string

Reply:
I added `.toString` because A value of type 'Object?' can't be assigned to a variable of type 'String'.
Try changing the type of the variable, or casting the right-hand type to 'String'.dart[invalid_assignment](https://dart.dev/diagnostics/invalid_assignment)
Object? value
Type: Object?